dll/win32/kernel32/kernel32_vista/EnumSystemLocalesEx.c

```diff
@@ -0,0 +1,51 @@
+#include "k32_vista.h"
+#include <winuser.h>
+
+struct enum_locale_ex_data
+{
+    LOCALE_ENUMPROCEX proc;
+    DWORD             flags;
+    LPARAM            lparam;
+};
+
+static BOOL CALLBACK enum_locale_ex_proc( HMODULE module, LPCWSTR type,
+                                          LPCWSTR name, WORD lang, LONG_PTR lparam )
+{
+    struct enum_locale_ex_data *data = (struct enum_locale_ex_data *)lparam;
+    WCHAR buffer[256];
+    DWORD neutral;
+    unsigned int flags;
+
+    GetLocaleInfoW( MAKELCID( lang, SORT_DEFAULT ), LOCALE_SNAME | LOCALE_NOUSEROVERRIDE,
+                    buffer, sizeof(buffer) / sizeof(WCHAR) );
+    if (!GetLocaleInfoW( MAKELCID( lang, SORT_DEFAULT ),
+                         LOCALE_INEUTRAL | LOCALE_NOUSEROVERRIDE | LOCALE_RETURN_NUMBER,
+                         (LPWSTR)&neutral, sizeof(neutral) / sizeof(WCHAR) ))
+        neutral = 0;
+    flags = LOCALE_WINDOWS;
+    flags |= neutral ? LOCALE_NEUTRALDATA : LOCALE_SPECIFICDATA;
+    if (data->flags && !(data->flags & flags)) return TRUE;
+    return data->proc( buffer, flags, data->lparam );
+}
+
+/******************************************************************************
+ *           EnumSystemLocalesEx  (KERNEL32.@)
+ */
+BOOL WINAPI EnumSystemLocalesEx( LOCALE_ENUMPROCEX proc, DWORD flags, LPARAM lparam, LPVOID reserved )
+{
+    struct enum_locale_ex_data data;
+    HMODULE kernel32_handle = GetModuleHandleW(L"kernel32.dll");
```

Comment:
If this code would live inside `kernel32.dll` could this be something like `&__ImageBase`, or does it always need to go via GMH?

Could this also be `kernelbase.dll` in future versions?

Reply:
This currently lives in kernel32_vista, so we need to use `GetModuleHandleW` here. The wine-10 kernelbase code doesn't use this anymore. (This is older code, but much easier to separate into it's own file. Eventually it will be replaced by the newer kernelbase code.)